File: backend/src/aithru_agent/stream/sse.py
import json

from .events import AgentStreamEvent


def format_sse_event(event: AgentStreamEvent) -> str:
    data = event.model_dump(mode="json")
    return f"id: {event.id}\nevent: {event.type}\ndata: {json.dumps(data, separators=(',', ':'))}\n\n"


def format_sse_comment(comment: str) -> str:
    safe_comment = comment.replace("\r", " ").replace("\n", " ")
    return f": {safe_comment}\n\n"

File: backend-ts/src/stream/sse.ts
import type { AgentStreamEvent } from "../contracts/types.js";

export function formatSseEvent(event: AgentStreamEvent): string {
  const data = JSON.stringify(event);
  return `id: ${event.id}\nevent: ${event.type}\ndata: ${data}\n\n`;
}

export function formatSseComment(comment: string): string {
  const safe = comment.replace(/\r/g, " ").replace(/\n/g, " ");
  return `: ${safe}\n\n`;
}
